Content.Server/Power/EntitySystems/PowerNetSystem.cs

```diff
@@ -297,25 +305,13 @@ private void ReconnectNetworks()
 
         private void UpdateApcPowerReceiver()
         {
-            var appearanceQuery = GetEntityQuery<AppearanceComponent>();
-            var metaQuery = GetEntityQuery<MetaDataComponent>();
-            var enumerator = AllEntityQuery<ApcPowerReceiverComponent>();
-            while (enumerator.MoveNext(out var uid, out var apcReceiver))
+            var enumerator = AllEntityQuery<ApcPowerReceiverComponent, MetaDataComponent>();
+            while (enumerator.MoveNext(out var uid, out var apcReceiver, out var metaDataComponent))
             {
-                var powered = apcReceiver.Powered;
-                if (powered == apcReceiver.PoweredLastUpdate)
+                if (metaDataComponent.EntityPaused)
                     continue;
 
-                if (metaQuery.GetComponent(uid).EntityPaused)
-                    continue;
```

Comment:
The reason it was using `AllEntityQuery` instead of just `EntityQuery` was so that it could defer getting the metadata comp unless it was actually required. There are quite a few power receiver entities, and in the majority of cases the power state doesn't change. So its probably better to only get the metadata comp after the `if (!entity.Comp1.Recalculate && entity.Comp1.Powered == powered)` check.

Reply:
Placed the component queries behind the check